### PR TITLE
Simplify group usage in poetry

### PR DESCRIPTION
**Pull request by fivegrant**
This PR
- Removes the `api` group and moves it into main.
- Factor out [testing dependencies into separate repo](https://github.com/jataware/service-test-tools)

ALSO JUST ADDED:
- Report uploads to S3
- GH Action that runs weekly (this might fail a few times until we get the env right)

### CODE REVIEW

File: env.sample [hunk 1]
```diff
@@ -7,4 +7,7 @@ MIT_TR_URL=http://mit-tr.staging.terarium.ai
 LOG_LEVEL=INFO
 MOCK_TA1=True
 MOCK_TDSt=True
-OPENAI_API_KEY=foo
\ No newline at end of file
+OPENAI_API_KEY=foo
```
Comment (YohannParis) [inline env.sample:10]: I removed the need to provide the open.ai key to MIT_TR, it is now embedded.

Comment (brandomr) [reply env.sample:10]: ah ok we may need to update some endpoints to remove it as a parameter if they no longer want it

File: tests/report.py [hunk 2]
```diff
@@ -42,9 +50,14 @@ def add_case(testobj):
             report[scenario]["description"] = spec["description"]
 
     timestamp = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
-    filename = f"tests/output/report_{timestamp}.json"
-    with open(filename, "w") as file:
+    filename = f"report_{timestamp}.json"
+    fullpath = os.path.join("tests/output", "filename")
```
Comment (brandomr) [inline tests/report.py:54]: should this be `integration-dashboard/knowledge-middleware/output/{report}`? I feel like `tests/output` is somewhat vague and we'll be using this bucket for testing for other TAs as well

**Review by brandomr — APPROVED** [review]

File: tests/report.py (hunk 2 above)
Comment (fivegrant) [reply tests/report.py:54]: we're writing it locally to tests/output/report.py but we actually writing to `ta1/report.py` on S3

File: env.sample (hunk 1 above)
Comment (YohannParis) [reply env.sample:10]: They still want it for sure. I just rewrote their service in our container to add it as an ENV variables on startup. We should do a clean up on TDS and KM [once we move to our image](https://github.com/DARPA-ASKEM/orchestration/pull/279)

**Pull request merged**